Assets/Scripts/Game/UserInterfaceWindows/DaggerfallInventoryWindow.cs

```diff
@@ -908,16 +940,80 @@ protected void AddLocalItem(DaggerfallUnityItem item)
         /// Creates filtered list of remote items.
         /// For now this just creates a flat list, as that is Daggerfall's behaviour.
         /// </summary>
-        protected virtual void FilterRemoteItems()
+        protected virtual void FilterRemoteItems(bool applyExtraFilter = true)
         {
+            DaggerfallUnityItem item;
             // Clear current references
             remoteItemsFiltered.Clear();
 
             // Add items to list
             if (remoteItems != null)
                 for (int i = 0; i < remoteItems.Count; i++)
-                    remoteItemsFiltered.Add(remoteItems.GetItem(i));
+                {
+                    if (applyExtraFilter)
+                    {
+                        item = remoteItems.GetItem(i);
+                        if (ItemPassesFilter(item))
+                            remoteItemsFiltered.Add(item);
+
+                    }
+                    else
+                        remoteItemsFiltered.Add(remoteItems.GetItem(i));
+                }
         }
+        ///<summary>
+        ///Checks whether inventory item meets criteria of filter
+        ///</summary>
+        protected virtual bool ItemPassesFilter(DaggerfallUnityItem item)
+        {
+            bool itemPasses = true;
+            bool iterationPass = false;
+
+            if (filterString.Length == 0)
+                return true;
+
+            foreach (string word in filterString.Split(' '))
+            {
+                if (word.Trim().Length > 0)
+                {
+                    if (word[0] == '-')
+                    {
+                        iterationPass = true;
+                        if (item.LongName.ToLower().Contains(word.Remove(0, 1)))
```

Comment:
There is a more performant way to do it:

```cs
if (item.LongName.IndexOf(word.Remove(0, 1), StringComparison.OrdinalIgnoreCase) != -1)
```

You should also store the result of `word.Remove(0, 1)` in a local variable instead of doing it twice.